### PR TITLE
feat(backdrop): change backdrop api to better handle positioning by p…

### PR DESCRIPTION
**Pull request by damienwebdev**
…arent components

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #118 

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
@lderrickable I took issue with the way you intertwined `z-index`s across different components. I think this PR solves the same goal, but allows us to circumvent that dependency. It does cause a change of the public API of a `daff-backdrop`, but I think it's an acceptable change as we're dropping `@Input` show for the more common `*ngIf`.

Let me know what you think.

**Comment by lderrickable** [comment]
could you fix the build and update the usage in demo?

**Comment by damienwebdev** [comment]
@lderrickable I fixed the docs and failing tests, but I couldn't find a distinct usage of the backdrop anywhere in demo. Everything using `daff-backdrop` seems to encapsulate the code inside of `@daffodil/design`.

File: libs/design/src/molecules/modal/modal/modal.component.spec.ts [hunk 1]
```diff
@@ -104,22 +98,31 @@ describe('DaffModalComponent', () => {
...
-  describe('on <backdrop>', () => {
+  describe('on <daff-backdrop>', () => {
 
     beforeEach(() => {
       backdrop = fixture.debugElement.query(By.css('daff-backdrop')).componentInstance;
     });
     
-    it('should set show to modal.show', () => {
-      expect(backdrop.show).toEqual(modal.show);
+    it('should remove add and remove the backdrop from the DOM as show changes', () => {
```
Comment (lderrickable) [inline libs/design/src/molecules/modal/modal/modal.component.spec.ts]: I think you have an extra "remove" here

File: libs/design/src/molecules/backdrop/backdrop/backdrop.component.spec.ts [hunk 2]
```diff
@@ -42,60 +42,31 @@ describe('DaffBackdropComponent', () => {
...
-
-  describe('backdrop transparency', () => {
+  describe('making the backdrop transparent, but still clickable', () => {
     beforeEach(() => {
       wrapper.backdropIsVisibleValue = false;
 
       fixture.detectChanges();
     });
 
     it('should apply the transparency class to the backdrop if `backdropIsVisible` is false', () => {
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply libs/design/src/molecules/backdrop/backdrop/backdrop.component.spec.ts]: Clarified in 118b9a823bdf10f9c9f2295096e3ffff95cb5f76

File: libs/design/src/molecules/modal/modal/modal.component.spec.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/design/src/molecules/modal/modal/modal.component.spec.ts]: Fixed in 27f8d1cab60c6f637d17d13cbf78d8072b565b8f

**Pull request merged**